semgrep-core/src/core/Pattern_match.ml

```diff
@@ -36,6 +36,27 @@
  * from a pattern:, we should not merge them!
  *)
 
+type tainted_tokens = Parse_info.t list [@@deriving show]
+
+(* Simplified version of Taint.source_to_sink meant for finding reporting *)
+type taint_call_trace =
+  (* A direct match *)
+  | Toks of Parse_info.t list
```

Comment:
why not tainted_tokens here?

Reply:
I believe Yoann asks why you are not using your type alias `tainted_tokens` instead of `Parse_info.t list`. And this `Parse_info.t list` is not the same as the "tainted tokens". This list of tokens represents a `Pattern_match.t`, whereas `tainted_tokens` are intermediate tokens (right now just variables) that are tainted.

Reply:
Clearly, I have already forgotten what code I wrote here :laughing:

After a second look here, I don't think that there was any particular reason that I didn't use `tainted_tokens` here. I think @IagoAbal's justification for this decision makes sense, though.

Reply:
Maybe you can add a type alias `pattern_match_tokens` and use it in the places where you are replacing a `Pattern_match.t` ?